src/common/data-modeler-service/ApplicationActions.ts

```diff
@@ -73,6 +111,21 @@ export class ApplicationActions extends DataModelerActions {
 
     this.databaseActionQueue.clearQueue(entityId);
 
+    if (entityType === EntityType.Model || entityType === EntityType.Table) {
+      // Clear existing profile actions in queue
+      const columns =
```

Comment:
There are multiple instances of this piece. Please move it to a separate method.